### DISCUSSION TITLE
docs/core/gi18n-example

### DISCUSSION THREAD

**Discussion by giscus** [gogf/gf-site#137] [General]
# docs/core/gi18n-example

在GoFrame框架中使用i18n国际化功能，包括对象创建、语言设置、常用方法以及与视图引擎的集成。详细描述了通过单例和独立对象管理语言转译的方式，使用SetLanguage和WithLanguage方法进行语言设定，并通过T和Tf方法实现关键字和模板内容的转译。文章还展示了通过上下文设置和视图引擎进行国际化操作的示例。

https://goframe.org/docs/core/gi18n-example

**Comment by nilsir** [comment]
gf:v2.8.2,    r.GetString undefined (type *ghttp.Request has no field or method GetString) 是不是换其他方法了

**Reply by gqcn** [reply]
感谢反馈，文档已修正。

**Comment by snowich** [comment]
在使用Tf方法时，goland编译器会报警告(格式字符串中没有占位符)，有什么好的解决办法吗

**Comment by damon35868** [comment]
请问如何在API的tag中使用？例如：
type LoginByCodeReq struct {
	g.Meta `path:"/user/loginByCode" method:"post" tags:"用户" summary:"用户注册"`
	Phone  string `json:"phone" v:"required-if:type,2|phone#{#ValidationPhoneRequired}|{#ValidationPhone}"`
	Email  string `json:"email" v:"required-if:type,1|email#{#ValidationEmailRequired}|{#ValidationEmail}"`
	Code   string `json:"code" v:"required|length:6,6#{#ValidationCodeRequired}|{#ValidationCodeLength}"`
	Type   int    `json:"type" v:"required|in:1,2#{#ValidationTypeRequired}|{#ValidationIn}"`
}

type LoginByCodeRes = UserLoginRes

**Comment by chaegumi** [comment]
这个多语言有问题吧，不是支持toml格式json格式吗
那下边的定义
[app]
site_name = "Site Name"

模板中不能这样用
{#app.site_name}
